src/core/thread/mle.cpp

```diff
@@ -246,6 +252,7 @@ ThreadError Mle::Stop(void)
         mNetif.RemoveUnicastAddress(mLeaderAloc);
     }
 
+    mNetif.GetPendingDataset().StopDelayTimer();
```

Comment:
Thanks a lot for your clarification.
For (i), I have updated the PR to stop the Delay Timer only if `Mle::Stop()` is triggered by `thread stop`;
For (ii), If the device has detached from the network on its own, we need to stop the current mNetwork's Delay Timer, and restart a new Timer for mLocal. I want to use another separate PR to address it, since it seems a separate feature enhancement, Do you agree?